Aydin/backend/src/course/course_controller.ts

```diff
@@ -0,0 +1,28 @@
+import {Router} from 'express';
+import {getManager} from 'typeorm';
+import {Course} from './course_entity';
+import {Lecturer} from '../lecturer/lecturer_entity';
+const CourseController = Router();
+CourseController.post('/:id', async (req, res) => {
+  try {
+    const course = new Course();
+    const lecturerRepo = getManager().getRepository(Lecturer);
+    const lecturer = await lecturerRepo.findOne({
+      where: {
+        id: req.params.id,
+      },
+    });
+    if (lecturer === undefined) {
```

Comment:
here i would double check that the API returns undefined for `findOne` and not null, for example, i would actually be a bit surprised as generally APIs tend to return null for empty values

Reply:
you may simply want to do `if(!lecturer)` yani

Reply:
i wanted to write clean code as clean as possible that why i didn't write if(!lecture) because i remembered you said the code should look stupid.

Reply:
if it is correct, then the way it is is fine, i'm simply asking to double check (or asking you to double check) that the api returns undefined. for example, i do not know this library, so i do not know if null is also a valid return value here, but many libs would return null here.

Reply:
yeah if api does not find anything matches in the database it  returns undefined